Xlib/ext/nvcontrol.py

```diff
@@ -0,0 +1,5358 @@
+    if not reply._data.get('flags'):
+        return None
+    return reply._data.get('list')
+
+
+def get_gpu_count(self):
+    """Return the number of GPU's present in the system."""
+    return int(query_target_count(self, Gpu()))
+
+
+def get_name(self, target):
+    """the GPU product name on which the specified X screen is running"""
+    return query_string_attribute(self, target, 0, NV_CTRL_STRING_PRODUCT_NAME)
+
+
+def get_driver_version(self, target):
+    """the NVIDIA (kernel level) driver version for the specified screen or GPU"""
+    return query_string_attribute(self, target, 0, NV_CTRL_STRING_NVIDIA_DRIVER_VERSION)
+
+
+def get_vbios_version(self, target):
+    """the version of the VBIOS for the specified screen or GPU"""
+    return query_string_attribute(self, target, 0, NV_CTRL_STRING_VBIOS_VERSION)
+
+
+def get_gpu_uuid(self, target):
+    return query_string_attribute(self, target, 0, NV_CTRL_STRING_GPU_UUID)
+
+
+def get_gpu_utilization(self, target):
+    string = query_string_attribute(self, target, 0, NV_CTRL_STRING_GPU_UTILIZATION)
+    result = {}
+    if string is not None and string != '':
+        for line in string.split(','):
+            [key, value] = line.split('=')[:2]
+            result[key.strip()] = int(value) if value.isdigit() else value
+    return result
+
+
+def get_performance_modes(self, target):
+    string = query_string_attribute(self, target, 0, NV_CTRL_STRING_PERFORMANCE_MODES)
+    result = []
+    if string is not None and string != '':
+        for perf in string.split(';'):
+            perf_dict = {}
+            for line in perf.split(','):
+                [key, value] = line.split('=')[:2]
+                perf_dict[key.strip()] = int(value) if value.isdigit() else value
+            result.append(perf_dict)
+    return result
+
+
+def get_vram(self, target):
+    return query_int_attribute(self, target, 0, NV_CTRL_VIDEO_RAM)
+
+
+def get_irq(self, target):
+    """Return the interrupt request line used by the GPU driving the screen"""
+    return query_int_attribute(self, target, 0, NV_CTRL_IRQ)
+
+
+def supports_framelock(self, target):
+    """returns whether the underlying GPU supports Frame Lock. All of the
+    other frame lock attributes are only applicable if this returns True."""
+    return query_int_attribute(self, target, 0, NV_CTRL_FRAMELOCK) == 1
+
+
+def gvo_supported(self, screen):
+    """returns whether this X screen supports GVO; if this screen does not
+    support GVO output, then all other GVO attributes are unavailable."""
+    return query_int_attribute(self, screen, [], NV_CTRL_GVO_SUPPORTED)
+
+
+def get_core_temp(self, target):
+    """return the current core temperature of the GPU driving the X screen."""
+    return query_int_attribute(self, target, 0, NV_CTRL_GPU_CORE_TEMPERATURE)
+
+
+def get_core_threshold(self, target):
+    """return the current GPU core slowdown threshold temperature. It
```

Comment:
The rest of docstrings should be corrected as well. Start it from capital letter, and put closing triple quotes on a separate line. Please use infinitive form of a verb everywhere.

Reply:
Hi @vasily-v-ryabov, I have fixed all the multiline doc strings, could you please take another look?